src/componnets/CardCart.js

```diff
@@ -0,0 +1,62 @@
+import { useState } from "react";
+import classes from "./Card.module.css";
+
+const CardCart = ({ productData, onDelete, onEditProductAmount }) => {
+  const [amount, setAmount] = useState(productData.amount);
+
+  const handleReduceClick = () => {
+    if (amount <= 1) {
+      onDelete(productData);
+    } else {
+      setAmount((prev) => prev - 1);
+      onEditProductAmount(productData, amount - 1);
+    }
+  };
+
+  const handleIncreaseClick = (event) => {
+    setAmount((prev) => prev + 1);
+    onEditProductAmount(productData, amount + 1);
+  };
+
+  const handleAmountChange = (event) => {
+    const newAmount = parseInt(event.target.value);
+    if (newAmount === 0 || newAmount === !NaN) {
```

Comment:
I guess the negating exclamation mark in front of `NaN` is by mistake. You probably want to check if amount is `NaN`. `!NaN` is actually just `true`.
```suggestion
    if (newAmount === 0 || newAmount === NaN) {
```